src/librustdoc/clean/utils.rs

```diff
@@ -303,7 +303,8 @@ pub(crate) fn name_from_pat(p: &hir::Pat<'_>) -> Symbol {
     debug!("trying to get a name from pattern: {p:?}");
 
     Symbol::intern(&match p.kind {
-        PatKind::Wild | PatKind::Struct(..) => return kw::Underscore,
+        // FIXME(never_patterns): does this make sense?
```

Comment:
What does rustdoc even do with patterns? 😆

Reply:
^^ I think it has something to do with printing function signatures but honestly idk

Reply:
oh, right, we need to render the pattern in:

```
fn foo(PATTERN: ty) {}
```

👍